docs/source/creating-interfaces/html-with-idom.rst

```diff
@@ -0,0 +1,84 @@
+HTML With IDOM
+==============
+
+In a typical Python-base web application the resonsibility of defining the view along
+with its backing data and logic are distributed between a client and server
+respectively. With IDOM, both these tasks are centralized in a single place. This is
+done by allowing HTML interfaces to be constructed in Python. Let's consider the HTML
+sample below:
+
+.. code-block:: html
+
+    <h1>My Todo List</h1>
+    <ul>
+        <li>Build a cool new app</li>
+        <li>Share it with the world!</li>
+    </ul>
+
+We can recreate the same thing with IDOM using functions attached to the
+:mod:`idom.html` module. These function share the same names as their corresponding HTML
+tags. For example, the `<h1/>` element above has a similarly named :func:`idom.html.h1`
+function. Given this we can write the following:
+
+.. testcode::
+
+    from idom import html
+
+    html.h1("My Todo List")
+    html.ul(
+        html.li("Build a cool new app"),
+        html.li("Share it with the world!"),
+    )
+
+That looks similar, but it's not very useful, we haven't captured the results from these
+function calls in a variable. To do this we need to wraps up layout above into a single
+:func:`idom.html.div` and assign it to a variable:
+
+.. testcode::
+
+    layout = html.div(
+        html.h1("My Todo List"),
+        html.ul(
+            html.li("Build a cool new app"),
+            html.li("Share it with the world!"),
+        ),
+    )
+
+Having done this we can inspect what is contained in our new ``layout`` variable. As it
+turns out, it holds a dictionary:
+
+.. testcode::
+
+    assert isinstance(layout, dict)
+
+And if we print the contents with :mod:`pprint` it should look like:
+
+.. testcode::
+
+    from pprint import pprint
+    pprint(layout, sort_dicts=False)
```

Comment:
useless `sort_dicts`

It doesn't really matter to the end user if the example is sorted or not.